### PR TITLE
cahute 0.3 (new formula)

### PR DESCRIPTION
**Pull request by seurd**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cahute.rb
Comment (SMillerDev) [inline Formula/c/cahute.rb:21]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/c/cahute.rb
Comment (SMillerDev) [inline Formula/c/cahute.rb:17]: Why install this one separately?

Comment (seurd) [reply Formula/c/cahute.rb:17]: It was accidentally left out of the CMakeLists.txt in release 0.2, fixed in [a commit right after](https://gitlab.com/cahuteproject/cahute/-/commit/157f710ccd99ac6eb2595dec136f0b4a93a8bcbb). We will remove that line in the next release.

Comment (SMillerDev) [reply Formula/c/cahute.rb:17]: Maybe just use the commit as a patch?

Comment (seurd) [reply Formula/c/cahute.rb]: I've done that now, but it causes `brew audit --new cahute` to fail.

File: Formula/c/cahute.rb
Comment (seurd) [reply Formula/c/cahute.rb:22]: I added a new test case for the library (compiling a test program from the documentation), but I'm not sure what the best way to go about testing the utilities is. If a calculator is connected, they would succeed, but if one isn't connected, they would fail. How would I turn that into a reproducible test?

File: Formula/c/cahute.rb
Comment (chenrui333) [inline Formula/c/cahute.rb:95]: looks like the project is fairly, can we ask for a new release? Thanks! (so that we dont need the patch)

Comment (seurd) [reply Formula/c/cahute.rb:95]: We can wait for the next release, yes. It should be out soon-ish.

Comment (chenrui333) [reply Formula/c/cahute.rb:95]: sounds good 👍

File: Formula/c/cahute.rb
Comment (SMillerDev) [reply Formula/c/cahute.rb:22]: Just assume none is connected. In most runs it will not be

**Comment by seurd** [comment]
Alright, it took a bit longer than expected, but now it's all done. This new version has no patch, slightly better tests, and some new dependencies needed for the updated release. Hope you like it!

File: Formula/c/cahute.rb
Comment (chenrui333) [inline Formula/c/cahute.rb:12]: we should vendor this out and log an ticket of removing setuptools usage. (but that can be followup)

File: Formula/c/cahute.rb
Comment (chenrui333) [inline Formula/c/cahute.rb:88]:
```suggestion
      *shell_output("pkg-config --cflags --libs cahute libusb-1.0").strip.split,
```

File: Formula/c/cahute.rb
Comment (chenrui333) [inline Formula/c/cahute.rb:12]:
```suggestion
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@seurd, thanks for your first contribution to homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**
